lib/core/text/test_abstract_text.nit

```diff
@@ -0,0 +1,34 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# This file is free software, which comes along with NIT.  This software is
+# distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without  even  the implied warranty of  MERCHANTABILITY or  FITNESS FOR A
+# PARTICULAR PURPOSE.  You can modify it is you want,  provided this header
+# is kept unaltered, and a notification of the changes is added.
+# You  are  allowed  to  redistribute it and sell it, alone or is a part of
+# another product.
+
+module test_abstract_text is test_suite
+
+import test_suite
+import flat
+
+redef class FlatText
+	# The examples in the documentation test the redefiniton in the (implicitly
+	# imported) `flat` module.
+	# Here, we force tests on the original definiton.
+
+	redef fun escape_to_c do return escape_to_c_text_impl
```

Comment:
You should not complexify an API for testing purpose.
I suggest that, in order to test the implementation of Text instead the ones of FlatText, to use a receiver that is a Text without being a FlatText.

Maybe something like:
~~~nit
intrude import core::ropes
var r = new Concat("??!??<", ??>??-")
assert r.escape_to_c == "?\\?!?\\?<?\\?>?\\?-"
~~~